Lib/_pyio.py

```diff
@@ -10,8 +10,13 @@
 import sys
 # Import _thread instead of threading to reduce startup cost
 from _thread import allocate_lock as Lock
-if sys.platform in {'win32', 'cygwin'}:
+if sys.platform == 'win32':
     from msvcrt import setmode as _setmode
+elif sys.platform == 'cygwin':
+    import ctypes
```

Comment:
We should avoid `ctypes` in the `io` module. There probably should cygwin equivalent for `msvcrt` instead.

Reply:
@benjaminp This is the _pyio module used primarily only in testing.  There is no "cygwin equivalent for `msvcrt`"--why should there be?  The closest I can think of is my own [PyCygwin](https://pycygwin.readthedocs.io/en/latest/) package, and it's not part of the standard lib, though I'd be willing to make it so; I'm hardly interested in fighting for a new stdlib package just to fix one bug though.

As was discussed [on the issue](https://bugs.python.org/issue28459#msg280203) it would be possible to expose Cygwin's `setmode()` as `os.setmode()` but it could be confusing since it does not function the same the FreeBSD `setmode()`.